### ISSUE TITLE
[BUG]: Testing SSH connection : New-Node failed with error Creation of Linux node failed

### ISSUE THREAD

**Issue by JohnVictorJV** [Azure/AKS-Edge#173]
**Describe the bug**
 Single machine deployment as per: https://learn.microsoft.com/en-us/azure/aks/hybrid/aks-edge-howto-single-node-deployment
 
**To Reproduce**
1. Run command '...'
   New-AksEdgeDeployment -JsonConfigFilePath .\aksedge-config.json
   
2. See error:
          - New-Node failed with error Creation of Linux node failed
          
          PS C:\dev\setup> New-AksEdgeDeployment -JsonConfigFilePath .\aksedge-config.install.json -verbose
         - Checking host for required features
         - Checking the status of 'Microsoft-Hyper-V'
         - Checking the status of 'Microsoft-Hyper-V-Management-PowerShell'
         - Checking the status of 'Microsoft-Hyper-V-Hypervisor'
         - Checking the status of 'OpenSSH.Client*'
         - Checking power management settings of the Host
         - Checking HNS version of the Host
         - Checking OpenSSH version of the Host
         - Checking Nested Virtualization of the Host
        
        [02/25/2024 21:28:47] All required host features are installed
        [02/25/2024 21:28:47] Validating AksEdge network parameters...
        [02/25/2024 21:28:47] *** 0 errors found in the deployment configuration.
        [02/25/2024 21:29:00] Checking the required certificates for offline installation...
        [02/25/2024 21:29:01] Verifying Host Requirements for Linux node(s)
         - Verifying host requirements for selected configuration (86.5 GB disk size, 33268 MB memory, 8 CPUs)
        [02/25/2024 21:29:01] Verifying Host OS can support requested configuration
        [02/25/2024 21:29:01] Verifying required storage, RAM and number of cores are available
         - Drive 'C:' has 549 GB free
         - A minimum of 86 GB disk space is required on drive 'C:'
         - Host has 54638 MB free memory
         - A minimum of 33268 MB memory is required
         - Host has 32 CPU cores
         - A minimum of 8 CPU cores is required
         - Verifying certificate requirements for AKS-EE
        VERBOSE: Perform operation 'Enumerate CimInstances' with following parameters, ''className' = Win32_OperatingSystem,'namespaceName' = root\cimv2'.
        VERBOSE: Operation 'Enumerate CimInstances' complete.
         - Starting Internet Disabled Deployment
        [02/25/2024 21:29:01] AksEdge - deploying a new Linux single machine k3s cluster
        [02/25/2024 21:29:02] Creating single machine cluster hcs network
         - AksEdge - private network carved:
        Name  : LinuxVmIp4Address
        Value : 192.168.0.2
        Name  : ip4GatewayAddress
        Value : 192.168.0.1
        Name  : ip4Subnet
        Value : 192.168.0.0
        Name  : ip4PrefixLength
        Value : 24
        Name  : WindowsVmIp4Address
        Value : 192.168.0.3
        Name  : ServiceIPRangeEnd
        Value : 192.168.0.4
        Name  : ServiceIPRangeStart
        Value : 192.168.0.4
        [02/25/2024 21:29:03] Deploying AKS Edge Essentials - K3s
        [02/25/2024 21:29:03] Step 1: Preparing host for AKS Edge Essentials - K3s
        [02/25/2024 21:29:03] Enabling Microsoft Update. This will allow AKS Edge Essentials - K3s to receive updates.
         - Microsoft Update is enabled.
        [02/25/2024 21:29:03] Checking for virtual switch with name 'aksedgesw-int'
         - The virtual switch 'aksedgesw-int' of type 'Internal' is present
        [02/25/2024 21:29:03] Associating wssdagent service with nodectl
        [02/25/2024 21:29:05] Creating vnet (name: aksedgesw-int)
         - Skipping DNS, not searching for DNS server
        [02/25/2024 21:29:06] Deploying AKS Edge Essentials - K3s
        [02/25/2024 21:29:06] Step 1: Skipped host preparation
        [02/25/2024 21:29:06] Step 2: Verifying AKS Edge Essentials - K3s installation
        [02/25/2024 21:29:06] Verifying installation
         - Verifying whether Hyper-V is enabled and functional
         - Hyper-V core services are active
         - Verifying expected Windows host binaries
         - Verifying expected Linux VM image
         - Testing for ssh key
         - Testing for wssdagent service
         - Testing if wssdagent is running
         - Testing if container resource is provisioned
         - Testing if vnet resource 'aksedgesw-int' is provisioned
        [02/25/2024 21:29:13] Step 3: Runtime configuration complete. Creating virtual machine
        [02/25/2024 21:29:13] Extracting Linux VHD
        [02/25/2024 21:29:17] Setting dynamically expanding virtual hard disk maximum size to 90.4 GB
         - Creating storage vhd (file: AzureIoTEdgeForLinux-v1-ledge)
         - Creating vnic (name: TestMachine-ledgeInterface)
         - Instantiating virtual machine (name: TestMachine-ledge)
         - Virtual machine successfuly instantiated
        [02/25/2024 21:29:25] Virtual machine created successfully.
        [02/25/2024 21:29:26] Successfully created virtual machine
        [02/25/2024 21:29:26] Virtual machine hostname: TestMachine-ledge
        [02/25/2024 21:29:26] Testing SSH connection...
         **New-Node failed with error Creation of Linux node failed**
       - Retrying...
       VERBOSE: Cleaning up partially created Linux VM
       [02/25/2024 21:30:57] Deleting virtual machine

**Expected behavior**

**Screenshots**

**Environment (please complete the following information):**
 - AKS Edge Essentials Version : AksEdge-K3s-1.27.6-1.6.384.0.msi
 - Windows Host OS (please complete the following information):
          -OS Name	Microsoft Windows 10 Enterprise
          - Version	10.0.19045 Build 19045
          - System Manufacturer	Dell Inc.
          - System Type	x64-based PC
          - System Model	Precision 7680
 - Powershell
       - --PS C:\dev\setup> $PSVersionTable
              Name                           Value
              ----                           -----
              PSVersion                      7.4.1
              PSEdition                      Core
              GitCommitId                    7.4.1
              OS                             Microsoft Windows 10.0.19045
              Platform                       Win32NT
              PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
              PSRemotingProtocolVersion      2.3
              SerializationVersion           1.1.0.1
              WSManStackVersion              3.0
              
**Additional context**
[aksedge-config.json](https://github.com/Azure/AKS-Edge/files/14400165/aksedge-config.json)
[aksedgelogs.zip](https://github.com/Azure/AKS-Edge/files/14400415/aksedgelogs.zip)

**Comment by logangeorge01** [comment]
Also seeing this

**Comment by logangeorge01** [comment]
I don't believe I was able to fix it even with multiple reinstallations of the aksee msi.

**Comment by matfax** [comment]
In my case, this was due to identity providers interfering with ssh. So I had to add `IdentityAgent=none` and `IdentitiesOnly=yes` to the `commandArray`s of `Invoke-WindowsNodeCommand`, `Invoke-EflowVmCommand`, `Copy-EflowVmFile`, and `Copy-WindowsVmFile` in the AksEdge psm1s. The cue to this was only printed in the logs, not in the stdout.

I would open a PR but the relevant functions don't seem to be exposed in the source.
